src/main/java/org/folio/rest/helper/TransactionSummariesHelper.java

```diff
@@ -37,7 +37,7 @@ public TransactionSummariesHelper(Map<String, String> okapiHeaders, Context ctx,
   }
 
   private void validateTransactionCount(Integer summaryCounts) {
-    if (summaryCounts < 1) {
+    if (summaryCounts < 0) {
```

Comment:
> 1. there are no encumbrances related to an invoice upon approval, but payments/credits must be created  upon invoice payment.
This will definitely happen

> 2. there are transactions that need to be updated upon invoice approval, no need to create payments/credits
This will also happen.  When an invoice is approved, if there are encumbrances associated with that invoice, the awaiting-payments API will be invoked updating those encumbrances (and budgets), moving money from encumbered to awaitingPayment.  It isn't until the invoice is paid that payments/credits are generated

Reply:
Thanks @craigmcnally. So I think, the change is in line with the requirement. @pautau-s let me know if we are missing any other case.

Reply:
> This will also happen. When an invoice is approved, if there are encumbrances associated with that invoice, the awaiting-payments API will be invoked updating those encumbrances (and budgets), moving money from encumbered to awaitingPayment. **It isn't until the invoice is paid that payments/credits are generated**

@craigmcnally Regarding the last sentence, currently we cannot update an invoiceTransactionSummary record after creation. That means we must specify both `numEncumbrances` and `numPaymentsCredits` upon approval. With this in mind, will the answers to the questions remain the same?

@javalkar-varun I believe that we should still prevent the creation of such records:
> * invoiceTransactionSummary
>   ```
>   {
>     "id": "5c156d82-96c3-4d9a-98e9-0c9a4184cf2c",
>     "numEncumbrances": 0,
>     "numPaymentsCredits": 0
>   }
>   ```
> * orderTransactionSummary
>   ```
>   {
>     "id": "5c156d82-96c3-4d9a-98e9-0c9a4184cf2c",
>     "numTransactions": 0
>   }
>   ```

Reply:
I don't see a problem with specifying both numEncumbrances and numPaymentsCredits when the invoice is approved.  This is actually what the wiki describes.  However, where the wiki is (currently) wrong is that it says both should be the same.  There are cases where that might not be the case - e.g. you have invoice-level adjustments that will result in payments that aren't also tied to an encumbrance.  I'll fix this on the wiki.

Remember that the invoice shouldn't be changing once approved, so you should be able to count up the fund distributions and figure out how many payments/credits will be needed, even though you're not generating payments yet.

Reply:
I think `numEncumbrances: 0` is technically valid, though may not come up often.  I'm not so sure about `numPaymentsCredits: 0` though...  What's the use case for that?  I don't see the point of having an invoice if there aren't any resulting payments or credits.

Reply:
So that means `orderTransactionSummary.numTransactions` and `invoiceTransactionSummary.numPaymentsCredits` must be greater than 0. And `invoiceTransactionSummary.numEncumbrances` >= 0

Reply:
Yes, I think that's correct @pautau-s